src/components/status-badges/external-dns-status-badge.tsx

```diff
@@ -0,0 +1,35 @@
+import { Icon } from '@equinor/eds-core-react';
+import { check, error_outlined, time } from '@equinor/eds-icons';
+import * as PropTypes from 'prop-types';
+import { FunctionComponent } from 'react';
+
+import {
+  StatusBadgeTemplate,
+  StatusBadgeTemplateProps,
+} from './status-badge-template';
+
+import { Tls } from '../../store/radix-api';
+
+type Status = Tls['status'];
+
+const BadgeTemplates: Record<
+  Status,
+  Pick<StatusBadgeTemplateProps, 'children' | 'icon' | 'type'>
+> = {
+  Pending: { type: 'warning', icon: <Icon data={time} /> },
+  Consistent: { icon: <Icon data={check} /> },
+  Invalid: { type: 'danger', icon: <Icon data={error_outlined} /> },
+};
+
```

Comment:
This is more type-safe, and easier to parse and read ;)

```ts
const BadgeTemplates = {
  Pending: { type: 'warning', icon: <Icon data={time} /> },
  Consistent: { icon: <Icon data={check} /> },
  Invalid: { type: 'danger', icon: <Icon data={error_outlined} /> },
} as const;
```
Record would allow BadgeTemplates to be modified with any string key, and requires us o test for it when using it.
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions

With that change this component gives a warning on`children`, since it doesnt actually exists on `BadgeTemplates`:
```jsx
export const ExternalDNSStatusBadge: FunctionComponent<{
  status: Status;
}> = ({ status }) => {
  const { children, ...rest } = BadgeTemplates[status];
  return (
    <StatusBadgeTemplate {...rest}>{children ?? status}</StatusBadgeTemplate>
  );
};
```

Reply:
I have made the record value readonly.

Reply:
keeping BadgeTemplates as-is

Reply:
fixed